scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,68 @@
+package scalafix.internal.reflect
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.core.Contexts.FreshContext
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.io.AbstractFile as DottyAbstractFile
+import dotty.tools.io.VirtualDirectory as DottyVirtualDirectory
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+import metaconfig.ConfError
+import metaconfig.Position
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  private val settings = "-unchecked" :: "-deprecation" :: "-classpath" :: classpath :: Nil
+  private val driver = new InteractiveDriver(settings)
+  private val reporter: StoreReporter = new StoreReporter()
+  private var ctx: FreshContext = driver.currentCtx.fresh
+  private val dottyVirtualDirectory = new DottyVirtualDirectory(target.name, None)
+  
+  ctx = ctx
+    .setReporter(reporter)
+    .setSetting(ctx.settings.outputDir, dottyVirtualDirectory)
```

Comment:
Rule compiler init is now fixed, the console manual test works:
```scala
scala> val r = new scalafix.internal.reflect.RuleCompiler(scalafix.internal.reflect.RuleCompilerClasspath.defaultClasspath)
val r: scalafix.internal.reflect.RuleCompiler = scalafix.internal.reflect.RuleCompiler@6c20679c
``` 
This was inspired by the following article with links to code samples:
- https://www.chris-kipp.io/blog/an-intro-to-the-scala-presentation-compiler
- https://github.com/com-lihaoyi/Ammonite/blob/master/amm/compiler/src/main/scala-3/ammonite/compiler/Compiler.scala
- https://github.com/scalameta/metals/blob/main/mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala